From 6b0f290c4ce0592d60cf0686460363a55d8cebd4 Mon Sep 17 00:00:00 2001
From: Konstantin Tokarev <annulen@yandex.ru>
Date: Wed, 16 Oct 2019 21:38:33 +0300
Subject: [PATCH] [cmake] Use release binaries for RelWithDebInfo and
 MinSizeRelease

It turns out that if build type of user project is not Release or
Debug, CMake picks first configuration from IMPORTED_CONFIGURATIONS
list. We have to move RELEASE to first place to match behavior of
other Qt modules, otherwise Windows users experience crashes caused by
runtimes mismtach.

Change-Id: I4329ea1b53b7ed5bee5fcbd57490736b57bccd9d
---
 Source/PlatformQt.cmake | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/Source/PlatformQt.cmake b/Source/PlatformQt.cmake
index c99d89d703f8..a2c84ee7b1c6 100644
--- a/Source/PlatformQt.cmake
+++ b/Source/PlatformQt.cmake
@@ -146,6 +146,18 @@ list(REMOVE_DUPLICATES Qt5@MODULE_NAME@_PRIVATE_INCLUDE_DIRS)
 list(REMOVE_DUPLICATES Qt5@MODULE_NAME@_DEFINITIONS)
 list(REMOVE_DUPLICATES Qt5@MODULE_NAME@_COMPILE_DEFINITIONS)
 list(REMOVE_DUPLICATES Qt5@MODULE_NAME@_EXECUTABLE_COMPILE_FLAGS)
+
+# Fixup order of configurations to match behavior of other Qt modules
+# See also https://bugreports.qt.io/browse/QTBUG-29186
+get_target_property(_configurations Qt5::@MODULE_NAME@ IMPORTED_CONFIGURATIONS)
+list(FIND _configurations RELEASE _index)
+if (\${_index} GREATER -1)
+    list(REMOVE_AT _configurations \${_index})
+    list(INSERT _configurations 0 RELEASE)
+    set_property(TARGET Qt5::@MODULE_NAME@ PROPERTY IMPORTED_CONFIGURATIONS \"\${_configurations}\")
+endif ()
+unset(_configurations)
+unset(_index)
 ")
 
 set(MODULE_NAME WebKit)
